console/console-init/ui/src/modules/device/dialogs/CreateDevice/AddGateway.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Title,
+  TextInput,
+  FormGroup,
+  Form,
+  Button,
+  FlexItem,
+  Flex,
+  Chip,
+  ChipGroup,
+  GridItem,
+  Grid
+} from "@patternfly/react-core";
+import { DeviceListAlert } from "modules/device/components";
+import { deviceIDRegExp } from "utils";
+
+export interface IGatewayProps {
+  addedGateways: string[];
+  removeGateway: (id: string) => void;
+  appendGateway: (deviceId: string) => void;
+}
+
+export const AddGateway: React.FunctionComponent<IGatewayProps> = ({
+  addedGateways,
+  removeGateway,
+  appendGateway
+}) => {
+  const [inputID, setInputID] = useState<string>();
+  const [isValid, setIsValid] = useState<boolean>(false);
+
+  const onHanldeClick = () => {
+    if (inputID) {
+      appendGateway(inputID);
+      setInputID("");
+    }
+  };
+
+  const alertDescription =
+    "In AMQ IoT, any existing device directly connected to cloud can turn" +
```

Comment:
Is there any reason for concatenating two string. If you want to split lines for readability purpose then i would like to suggest use '~' syntax for multi line.
const alertDescription=`
In AMQ IoT, any existing device directly connected to cloud can turn
to be a gateway by been added to other devices for their connections. Multiple gateways are supported.
`